packages/fast-usdc/src/type-guards.js

```diff
@@ -0,0 +1,28 @@
+import { M } from '@endo/patterns';
+
+/**
+ * @import {TypedPattern} from '@agoric/internal'
+ * @import {USDCProposalShapes} from './pool-share-math'
+ */
+
+/**
+ * @param {Brand} brand must be a 'nat' brand, not checked
+ * @param {NatValue} [min]
+ */
+export const makeNatAmountShape = (brand, min) =>
+  harden({ brand, value: min ? M.gte(min) : M.nat() });
+
+/** @param {Record<'PoolShares' | 'USDC', Brand<'nat'>>} brands */
+export const makeProposalShapes = ({ PoolShares, USDC }) => {
+  /** @type {TypedPattern<USDCProposalShapes['deposit']>} */
+  const deposit = M.splitRecord(
+    { give: { USDC: makeNatAmountShape(USDC, 1n) } },
```

Comment:
Beneficial to parameterize `min` here?

Reply:
(anticipated) product req: user must deposit minimum of `1e6` or `10e6` units, not 1n units, so initial share price is not weird

Reply:
weird how?

note that share worth starts 1-1:

```js
      const dust = AmountMath.make(USDC, 1n);
      const shareWorth = makeParity(dust, PoolShares);
```

Other than that, there are no special cases, AFAICT.